Userland/Utilities/cat.cpp

```diff
@@ -37,13 +41,46 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     TRY(Core::System::pledge("stdio"));
 
+    size_t line_count = 1;
     Array<u8, 32768> buffer;
     for (auto const& file : files) {
         while (!file->is_eof()) {
             auto const buffer_span = TRY(file->read_some(buffer));
-            out("{:s}", buffer_span);
+            if (show_lines) {
+                line_count = output_buffer_with_line_numbers(line_count, buffer_span);
+            } else {
+                out("{:s}", buffer_span);
+            }
         }
+        line_count = 1;
     }
 
     return files.size() != paths.size();
 }
+
+size_t output_buffer_with_line_numbers(size_t line_count, AK::Span<unsigned char> buffer_span)
+{
+    auto buffer_itter = buffer_span.begin();
+    while (!buffer_itter.is_end()) {
+        // to know if the next itter is the end
+        auto curr_value = *buffer_itter;
+        buffer_itter++;
+
+        if ((curr_value != '\n') && (line_count != 1)) {
+            out("{:c}", curr_value);
+        } else {
+            if (line_count == 1) {
+                out("   {}\t{:c}", line_count, curr_value);
+            } else {
+                out("{:c}", curr_value);
+                if (!buffer_itter.is_end()) {
+                    // This matches the formatting of gnu's cat function
+                    out("   {:d}\t", line_count);
```

Comment:
This won't produce the right output, the number of space before the line number needs to vary.
As an example, this is an extract of `cat -n README.md` on Linux:
```
    98	* **Tom** - [tomuta](https://github.com/tomuta)
    99	* **Paul Scharnofske** - [asynts](https://github.com/asynts)
   100	* **Itamar Shenhar** - [itamar8910](https://github.com/itamar8910)
   101	* **Luke Wilde** - [Lubrsi](https://github.com/Lubrsi)
```
Off the top of my head, the correct formatting is something like:
`out("{:06}\t", line_count);`

Reply:
The correct format is `"{: >6}\t"`